lib/text_diff.ml

```diff
@@ -482,18 +466,25 @@ and process_tuple_type_diff tuple_diff =
             Iconflict
               { iorig = Some (star ^ type_expr_to_string te); inew = None };
           ]
-      | Changed (Modified { reference; current }) ->
-          let te_hunk =
-            Iconflict
-              {
-                iorig = Some (type_expr_to_string reference);
-                inew = Some (type_expr_to_string current);
-              }
-          in
-          if i > 0 then [ Icommon " * "; te_hunk ] else [ te_hunk ])
+      | Changed (Modified te) ->
+          let te_hunks = process_type_expr_diff te in
+          if i > 0 then Icommon " * " :: te_hunks else te_hunks)
     tuple_diff
   |> List.concat
 
+and process_type_expr_diff (diff : Diff.typ_exp) : inline_hunk list =
+  match diff with
+  | Diff.Atomic { reference; current } ->
+      [
+        Iconflict
+          {
+            iorig = Some (type_expr_to_string reference);
+            inew = Some (type_expr_to_string current);
+          };
+      ]
+  | Tuple tuple_diff ->
+      (Icommon "(" :: process_tuple_type_diff tuple_diff) @ [ Icommon ")" ]
```

Comment:
I don't think that's the right place for adding brackets or at least not unconditionally, the caller should probably be responsible for this.

This could be controlled by a labelled arg for instance!